atomic_reactor/pulp_util.py

```diff
@@ -224,3 +227,55 @@ def get_registry_hostname(self):
 
     def get_pulp_instance(self):
         return self.pulp_instance
+
+
+class LockedPulpRepository(object):
+    """
+    Context manager for Pulp repository operations.
+
+    This context manager creates a 'lock' repository on entry, and
+    deletes the repository on exit.
+
+    Usage:
+
+    with LockedPulpRepository(handler, repo_id) as locked_repo:
+        locked_repo.set_image_tags(v1_image_id, tags)
+    """
+
+    def __init__(self, pulp, repo_id, prefix='lock-'):
+        self.pulp = pulp
+        self.repo_id = repo_id
+        self.prefix = prefix
+
+        self.retry_times = 9
+        self.retry_delay = 5
+
+    def __enter__(self):
+        logger.info("creating lock repository %s%s", self.prefix, self.repo_id)
+        registry_id = 'lock/{}'.format(self.repo_id)
+        retry = 0
+        for counter in range(self.retry_times + 1):
```

Comment:
This looked weird to me until I realize you're implementing a do while loop and need to loop one extra time for the hopefully successful initial try. Still, it's not an idiom we use consistently inside the code base so would you please add a comment explaining why the range is retry_times + 1?

Reply:
Added.